src/transformers/models/m2m_100/configuration_m2m_100.py

```diff
@@ -153,3 +159,126 @@ def __init__(
             decoder_start_token_id=decoder_start_token_id,
             **kwargs,
         )
+
+
+class M2M100OnnxConfig(OnnxSeq2SeqConfigWithPast):
+    @property
+    def inputs(self) -> Mapping[str, Mapping[int, str]]:
+        common_inputs = OrderedDict(
+            [
+                ("input_ids", {0: "batch", 1: "encoder_sequence"}),
+                ("attention_mask", {0: "batch", 1: "encoder_sequence"}),
+            ]
+        )
+
+        if self.use_past:
+            common_inputs["decoder_input_ids"] = {0: "batch"}
+            common_inputs["decoder_attention_mask"] = {0: "batch", 1: "past_decoder_sequence + sequence"}
+        else:
+            common_inputs["decoder_input_ids"] = {0: "batch", 1: "decoder_sequence"}
+            common_inputs["decoder_attention_mask"] = {0: "batch", 1: "decoder_sequence"}
+
+        if self.use_past:
+            self.fill_with_past_key_values_(common_inputs, direction="inputs")
+        return common_inputs
+
+    # Copied from BartOnnxConfig._generate_dummy_inputs_for_sequence_classification_and_question_answering
+    # A better name would be _generate_dummy_inputs_for_encoder_and_decoder because sequence classification and question
```

Comment:
Yes but with the way it's done for MarianMt, you cannot put the `# Copied from` above `_generate_dummy_inputs_for_default_and_seq2seq_lm` so you will not be notified if it changes, and [it might due to the todo left](https://github.com/huggingface/transformers/blob/531336bbfd2a97cf800f610d971d6ec0a1578752/src/transformers/models/marian/configuration_marian.py#L292).

Reply:
Thanks for the clarification! 

My suggestion would then be to accept your current `# Copied from` approach, and then in a separate PR we think about refactoring this potentially confusing method name in BART. That way, all derived models don't have to include this warning message - wdyt?

Reply:
Sounds good!